File: q0287/q0287.cpp
#include <vector>
using namespace std;

class Solution
{
public:
    int findDuplicate(vector<int> &nums)
    {
        for (auto num : nums)
        {
            int idx = abs(num);
            if (nums[idx - 1] < 0)
            {
                return idx;
            }
            nums[idx - 1] *= -1;
        }
        return -1;
    }
};

File: q0287/q0287.java
class Solution {
    public int findDuplicate(int[] nums) {
        for (int num : nums) {
            int idx = Math.abs(num);
            if (nums[idx - 1] < 0) {
                return idx;
            }
            nums[idx - 1] *= -1;
        }
        return -1;
    }
}
